wp-web-push/lib/delapuente/wp-sw-manager/class-wp-sw-manager.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+require_once(__DIR__ . '/class-wp-sw-manager-router.php');
+require_once(__DIR__ . '/class-wp-sw-manager-combinator.php');
+
+/**
+ * Holds the shared manager for composing the service workers.
+ *
+ * Service Workers enable web applications to send push notifications, work
+ * offline or perform background tasks periodically. Currently the standard only
+ * allows **one service worker per scope** making it hard for plugin developers
+ * to combine different focused and isolated functionality.
+ *
+ * The WP_SW_Manager library provides a collaborative way to generate service
+ * workers. It is as simple as registering a callback for writing your service
+ * worker functionality:
+ *
+ * ```php
+ * WP_SW_Manager::get_manager()->sw()->add_contents(write_sw);
+ *
+ * function write_sw() {
+ *     echo 'console.log("Here is my plugin!")';
+ * }
+ * ```
+ *
+ * @version 0.1.0
+ * @license GPL
+ * @author Salvador de la Puente González <salva@unoyunodiez.com>
+ */
+class WP_SW_Manager {
+    /**
+     * The name of the enqueued script in charge of registering the
+     * service workers.
+     *
+     * In case you have client code depending on the registration of
+     * a service worker, use this const as dependency to ensure your
+     * script is added **after** the registration script:
+     *
+     * ```php
+     * register_script('my-plugin-script', 'url/to/my/script.js', array(WP_SW_Manager::SW_REGISTRAR_SCRIPT));
+     * ```
+     *
+     * From your JavaScript code, you can access these registrations
+     * via `$swRegistrations` variable.
+     *
+     * @see WP_SW_Manager::get_js_id()
+     * @api
+     * @var string
+     */
+    const SW_REGISTRAR_SCRIPT = 'wp-sw-manager-registrar';
+
+    const SW_REGISTRAR_SCRIPT_URL = 'wpswmanager/sw-registrar.js';
```

Comment:
Yes, I didn't notice that the documentation (http://php.net/manual/en/language.oop5.constants.php) says only constant expressions are available since 5.6.